browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,342 @@
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+content::WebContents* SplitViewLocationBar::GetActiveWebContents() const {
+  return web_contents();
+}
+
+bool SplitViewLocationBar::ShouldDisplayURL() const {
+  content::NavigationEntry* entry = GetNavigationEntry();
+  if (entry && !entry->IsInitialEntry()) {
+    // We don't want to hide chrome://newtab url for this location bar.
+    const auto is_ntp = [](const GURL& url) {
+      return url.SchemeIs(content::kChromeUIScheme) &&
+             url.host() == chrome::kChromeUINewTabHost;
+    };
+    if (is_ntp(entry->GetVirtualURL()) || is_ntp(entry->GetURL())) {
+      return true;
+    }
+  }
+
+  return ChromeLocationBarModelDelegate::ShouldDisplayURL();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  auto url_text = GetURLForDisplay();
+  if (HasCertError()) {
+    // In case of cert error, we show https scheme with another view so that
+    // we can strike the scheme. So we omit HTTPS here.
+    https_with_strike_->SetVisible(true);
+    scheme_separator_->SetVisible(true);
+    if (url_text.starts_with(u"https://")) {
+      url_text = url_text.substr(8);
+    } else {
+      CHECK_IS_TEST();
+    }
+  } else {
+    https_with_strike_->SetVisible(false);
+    scheme_separator_->SetVisible(false);
+  }
+
+  url_->SetText(url_text);
+  UpdateIcon();
+
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::UpdateIcon() {
+  // At the moment, we show only warning icon.
+  safety_icon_->SetVisible(!IsContentsSafe());
+}
+
+bool SplitViewLocationBar::IsContentsSafe() const {
+  auto security_level = location_bar_model_->GetSecurityLevel();
+  return security_level == security_state::SecurityLevel::SECURE ||
+         security_level == security_state::SecurityLevel::NONE;
+}
+
+bool SplitViewLocationBar::HasCertError() const {
+  return net::IsCertStatusError(location_bar_model_->GetCertStatus());
+}
+
+std::u16string SplitViewLocationBar::GetURLForDisplay() const {
+  auto url = location_bar_model_->GetURLForDisplay();
+  if (location_bar_model_->GetURL().scheme() == url::kHttpScheme &&
```

Comment:
Do we care that we're going to show it on `localhost` urls? I think we generally treat them as secure ish

Reply:
Let's hear about it from security team. cc. @thypon

Reply:
I don't think we should be modifying the display url, the whole point of using the location bar model is to ensure that it always displays correctly.

Reply:
if for some reason we want to display it differently than it would be displayed in the urlbar then we should use `GetFormattedURL`.

Reply:
I think we're intentionally doing something different to the LocationBar here:

This is the LocationBar on `http.badssl.com`
![image](https://github.com/user-attachments/assets/8116a4fe-4d68-46ba-881c-ff847a0926be)


Here I think we're deliberately including the `http://` to make it more obvious whats wrong (we don't show the big `Not secure` text in the split view url bar)
![image](https://github.com/user-attachments/assets/eb90993b-d6a1-4614-bc95-9f051f7ff1c4)

Reply:
@bcaller , @thypon Could you tell us your thoughts on this? We can either make difference from omnibox or make it same.

Reply:
We talked in DM and decided to match omnibox's behavior

Reply:
I'm fine with however we want to display it, I just wanted to make sure we were using the right method if we wanted a different format